Update LeakyRelu.pbtxt

01-Abdul-Hadi opened the pull request:
https://github.com/tensorflow/tensorflow/issues/60521 issue no. #60521
"According to doc(https://tensorflow.google.cn/api_docs/python/tf/nn/leaky_relu), the argument features can be float16, float32, float64, int32, int64. But the error message in following snippet code indicates that the type of features can only be float16, float32, float64 without int." This is my first contribution to open source so please excuse any mistake and please tell if anything is wrong.

01-Abdul-Hadi committed:
Update LeakyRelu.pbtxt

https://github.com/tensorflow/tensorflow/issues/60521
issue no. #60521
"According to doc(https://tensorflow.google.cn/api_docs/python/tf/nn/leaky_relu), the argument features can be float16, float32, float64, int32, int64. But the error message in following snippet code indicates that the type of features can only be float16, float32, float64 without int."
This is my contribution to open source so please excuse any mistake and please tell if anything is wrong.

mihaimaruseac commented:
Please don't use "add file"/"update file"/"fix file"/etc. commit messages. These are hard to reason about when looking at the history of the file/repository. Instead, please write explanatory git commit messages.

The commit message is also the title of the PR if the PR has only one commit. It is thus twice important to have commit messages that are relevant, as PRs would be easier to understand and easier to analyze in search results.

For how to write good quality git commit messages, please consult https://cbea.ms/git-commit/

mihaimaruseac reviewed (changes requested):
This is only the v1 version of the OP. You actually have to change V2. Plus, you need to change in C++ code, this is generated from that.

01-Abdul-Hadi commented:
Hi,
       I am grateful to you for informing me of my mistake and also giving
me the link to the git commit webpage.
       This is my first contribution to open source and so I dont know
anything about accepted standards of commiting and other things.
       Should I continue to work on my pull request or should I close it as
I think it is too late and also I dont want to cause anymore inconvenience
for the tensorflow GitHub and also for moderators like you?

Thanking you,
Abdul Hadi

On Sat, 6 May, 2023, 8:56 pm Mihai Maruseac, ***@***.***>
wrote:

> Please don't use "add file"/"update file"/"fix file"/etc. commit messages.
> These are hard to reason about when looking at the history of the
> file/repository. Instead, please write explanatory git commit messages.
>
> The commit message is also the title of the PR if the PR has only one
> commit. It is thus twice important to have commit messages that are
> relevant, as PRs would be easier to understand and easier to analyze in
> search results.
>
> For how to write good quality git commit messages, please consult
> https://cbea.ms/git-commit/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/60522#issuecomment-1537164737>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AV767AIGTOE7LB6VOUGV56LXEZURTANCNFSM6AAAAAAXYA26SY>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

01-Abdul-Hadi committed:
add int value in allowed_values list 

if applied, this commit will add DT_INT as a type of value in allowed_values list in LeakyRelu.pbtxt file.

swachhandl reviewed (commented):
Were you able to try whether this worked after you made your changes? 

I think you'll need to change the types in the op registration here: https://github.com/tensorflow/tensorflow/blob/3efa230babd2dcd9ac6fdc50246a1afc1f96a0d9/tensorflow/core/ops/nn_ops.cc#L1219 

Acc. to https://github.com/tensorflow/tensorflow/pull/60522#pullrequestreview-1415788956, it seems like the .pbtxt files are autogenerated.  

It also looks like we have kernels for integral types for CPU: https://github.com/tensorflow/tensorflow/blob/3efa230babd2dcd9ac6fdc50246a1afc1f96a0d9/tensorflow/core/kernels/relu_op.cc#L61 

But not for GPU (integral types are usually not supported on GPUs). See https://github.com/tensorflow/tensorflow/blob/24f323096dd48615ca59c208708ca2cb8d88685f/tensorflow/core/kernels/relu_op.cc#L217 and 
https://github.com/tensorflow/tensorflow/blob/2e86aa79d38698d43e9aebdeab55b32ddcaa819c/tensorflow/core/framework/register_types.h#L204-L205 

Please add unit tests for the changes you'll be making. Thanks!

swachhandl commented:
@cantonios Is my understanding above correct? I'm guessing we'd need to only allow integral types for CPU but not for GPU. Is that possible to do in the same op registration or do we need to have separate op registrations if we want to do this? 

Are there use cases where LeakyRelu is needed to work for integral types? If not, do you think we should just modify the documentation?

cantonios commented:
> @cantonios Is my understanding above correct? I'm guessing we'd need to only allow integral types for CPU but not for GPU. Is that possible to do in the same op registration or do we need to have separate op registrations if we want to do this?

Your understanding is correct that we need to update the op Attrs to include the type.  If registrations don't exist, you might want to add those as well to actually test it.  You'll need to ensure the kernels will actually work for integers - I haven't looked at the implementation to check.  Otherwise you may need to write new ones.  If it's not that necessary for GPU, it's okay to leave out those kernels/registrations - it would just end up calling the CPU versions (and lead to some host-device data transfers for GPU models).

> Are there use cases where LeakyRelu is needed to work for integral types? If not, do you think we should just modify the documentation?

Maybe for integer-quantized models?  But I generally doubt it.  It's probably easier to update the documentation.

gbaned commented:
Hi @swachhandl Any update on this PR? Please. Thank you!

gbaned commented:
Hi @swachhandl Any update on this PR? Please. Thank you!

gbaned commented:
Hi @swachhandl Any update on this PR? Please. Thank you!

swachhandl commented:
Sorry I think I didn't get notified for the above pings. I don't have anything to add, awaiting the PR author's reply to my comment above.

gbaned commented:
Hi @01-Abdul-Hadi Can you please check @swachhandl's comments and keep us posted ? Thank you!

gbaned commented:
Hi @01-Abdul-Hadi Any update on this PR? Please. Thank you!
